addons/docs/src/frameworks/common/enhanceArgTypes.test.ts

```diff
@@ -0,0 +1,102 @@
+import yaml from 'yaml';
+import { ArgType } from '@storybook/api';
+import { enhanceArgTypes } from './enhanceArgTypes';
+
+expect.addSnapshotSerializer({
+  print: (val: any) => yaml.stringify(val).trim(),
+  test: (val) => true,
+});
+
+const enhance = (argType: ArgType, arg?: any, extraction?: ArgType) => {
+  const context = {
+    id: 'foo--bar',
+    kind: 'foo',
+    name: 'bar',
+    parameters: {
+      component: 'dummy',
+      docs: {
+        extractArgTypes: () => ({ input: extraction }),
+      },
+      argTypes: {
+        input: argType,
+      },
+    },
+    args: {
+      input: arg,
+    },
+    globalArgs: {},
+  };
+  const result = enhanceArgTypes(context);
+  return result?.argTypes?.input;
+};
+
+describe('enhanceArgTypes', () => {
+  describe('argTypes input', () => {
+    it('number', () => {
+      expect(
+        enhance({
+          type: { name: 'number' },
+        })
+      ).toMatchInlineSnapshot(`
+        name: input
+        type:
+          name: number
+      `);
+    });
+  });
+  describe('args input', () => {
+    it('number', () => {
+      expect(enhance({}, 5)).toMatchInlineSnapshot(`name: input`);
```

Comment:
Shouldn't this have `type: { name: 'number' }`?

Reply:
This is a bug. Have updated the code. The enhancer code was looking at `context.parameters.args` and the test was passing `context.args`.

Pretty sure I wrote it that way for a reason, and it might show more clearly during the controls PR.